test/conformance/exp_command_buffer/update/kernel_handle_update.cpp

```diff
@@ -0,0 +1,347 @@
+        std::shared_ptr<std::vector<char>> ILBinary;
+        std::vector<ur_program_metadata_t> Metadatas{};
+
+        ur_platform_backend_t Backend;
+        ASSERT_SUCCESS(urPlatformGetInfo(Platform, UR_PLATFORM_INFO_BACKEND,
+                                         sizeof(Backend), &Backend, nullptr));
+
+        ASSERT_NO_FATAL_FAILURE(
+            uur::KernelsEnvironment::instance->LoadSource(Name, ILBinary));
+
+        const ur_program_properties_t Properties = {
+            UR_STRUCTURE_TYPE_PROGRAM_PROPERTIES, nullptr,
+            static_cast<uint32_t>(Metadatas.size()),
+            Metadatas.empty() ? nullptr : Metadatas.data()};
+        ASSERT_SUCCESS(uur::KernelsEnvironment::instance->CreateProgram(
+            Platform, Context, Device, *ILBinary, &Properties, &Program));
+
+        auto KernelNames =
+            uur::KernelsEnvironment::instance->GetEntryPointNames(Name);
+        std::string KernelName = KernelNames[0];
+        ASSERT_FALSE(KernelName.empty());
+
+        ASSERT_SUCCESS(urProgramBuild(Context, Program, nullptr));
+        ASSERT_SUCCESS(urKernelCreate(Program, KernelName.data(), &Kernel));
+    }
+
+    virtual void setUpKernel() = 0;
+
+    virtual void destroyKernel() {
+        ASSERT_SUCCESS(urKernelRelease(Kernel));
+        ASSERT_SUCCESS(urProgramRelease(Program));
+    };
+
+    virtual void validate() = 0;
+
+    std::string Name;
+    ur_platform_handle_t Platform;
+    ur_context_handle_t Context;
+    ur_device_handle_t Device;
+    ur_program_handle_t Program;
+    ur_kernel_handle_t Kernel;
+};
+
+struct TestSaxpyKernel : public TestKernel {
+
+    TestSaxpyKernel(ur_platform_handle_t Platform, ur_context_handle_t Context,
+                    ur_device_handle_t Device)
+        : TestKernel("saxpy_usm", Platform, Context, Device) {}
+
+    ~TestSaxpyKernel() override = default;
+
+    void setUpKernel() override {
+
+        ASSERT_NO_FATAL_FAILURE(buildKernel());
+
+        const size_t AllocationSize = sizeof(uint32_t) * GlobalSize;
+        for (auto &SharedPtr : Memory) {
+            ASSERT_SUCCESS(urUSMSharedAlloc(Context, Device, nullptr, nullptr,
+                                            AllocationSize, &SharedPtr));
+            ASSERT_NE(SharedPtr, nullptr);
+
+            std::vector<uint8_t> pattern(AllocationSize);
+            uur::generateMemFillPattern(pattern);
+            std::memcpy(SharedPtr, pattern.data(), AllocationSize);
+        }
+
+        // Index 0 is the output
+        ASSERT_SUCCESS(urKernelSetArgPointer(Kernel, 0, nullptr, Memory[0]));
+        // Index 1 is A
+        ASSERT_SUCCESS(urKernelSetArgValue(Kernel, 1, sizeof(A), nullptr, &A));
+        // Index 2 is X
+        ASSERT_SUCCESS(urKernelSetArgPointer(Kernel, 2, nullptr, Memory[1]));
+        // Index 3 is Y
+        ASSERT_SUCCESS(urKernelSetArgPointer(Kernel, 3, nullptr, Memory[2]));
+    }
+
+    void destroyKernel() override {
+        for (auto &shared_ptr : Memory) {
```

Comment:
nitpick - there's a few places for here on where the variable naming convention changes
```suggestion
        for (auto &SharedPtr : Memory) {
```